spanner/tests/system/test_system.py

```diff
@@ -488,6 +488,35 @@ def test_transaction_read_and_insert_then_rollback(self):
         self.assertEqual(rows, [])
 
     @RetryErrors(exception=GrpcRendezvous)
+    def test_transaction_read_and_insert_then_execption(self):
+        retry = RetryInstanceState(_has_all_ddl)
+        retry(self._db.reload)()
+
+        class TestException(Exception):
+            pass
+
+        session = self._db.session()
+        session.create()
+        self.to_delete.append(session)
+
+        with session.batch() as batch:
+            batch.delete(self.TABLE, self.ALL)
+
+        def _unit_of_work(transaction):
```

Comment:
Figuring out what is in-scope and out-of-scope in this closure is painful to "sight read". Instead, please use an instance method (which will still have access to `self.TABLE`, etc.)